lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
I depends how it is looked at. For me `AtomicFixnum` means it can only hold `Fixnums` so no overflow supported. If it was supported, the name would be `AtomicInteger`.